src/context/chat-context.tsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext, useState, useMemo } from 'react'
+import { Offer } from '~/types'
+
+interface ChatProviderProps {
+  children: React.ReactNode
+}
+
+interface ChatProvideContext {
+  chatInfo: Pick<Offer, 'author' | 'authorRole' | 'chatId'> | null
+  setChatInfo: (
+    chatInfo: Pick<Offer, 'author' | 'authorRole' | 'chatId'> | null
+  ) => void
+}
+
+const ChatContext = createContext<ChatProvideContext>({} as ChatProvideContext)
+
+const ChatProvider = ({ children }: ChatProviderProps) => {
+  const [chatInfo, setChatInfo] = useState<Pick<
+    Offer,
+    'author' | 'authorRole' | 'chatId'
```

Comment:
you are using this Pick from Offer 3 times, better create a type or interface

Reply:
Fixed